libs/x-content/src/main/java/org/elasticsearch/xcontent/DotExpandingXContentParser.java

```diff
@@ -48,7 +48,7 @@ public Token nextToken() throws IOException {
 
         private void expandDots() throws IOException {
             String field = delegate().currentName();
-            String[] subpaths = field.split("\\.");
+            String[] subpaths = splitAndValidatePath(field);
             if (subpaths.length == 0) {
```

Comment:
@romseygeek this check is redundant now?

Reply:
It is, good catch